src/librustc_middle/ty/print/pretty.rs

```diff
@@ -572,7 +572,14 @@ pub trait PrettyPrinter<'tcx>:
                     let mut is_sized = false;
                     p!(write("impl"));
                     for predicate in bounds.predicates {
-                        if let Some(trait_ref) = predicate.to_opt_poly_trait_ref() {
+                        // Note: We can't use `to_opt_poly_trait_ref` here as `predicate`
+                        // may contain unbound variables. We therefore do this manually.
+                        //
+                        // FIXME(lcnr): Find out why exactly this is the case :)
+                        if let ty::PredicateAtom::Trait(pred, _) =
+                            predicate.bound_atom(self.tcx()).skip_binder()
```

Comment:
isn't this the same as `skip_binders`? But yeah I'm not quite sure why `to_opt_poly_trait_ref` doesn't work...

Reply:
it was, I want to check in `skip_binders` that there are no unbounds vars, which would fail here and cause an ICE